chain/substreams/src/chain.rs

```diff
@@ -148,3 +140,97 @@ impl Blockchain for Chain {
         unimplemented!()
     }
 }
+
+pub struct Mapper {}
+
+#[async_trait]
+impl SubstreamsMapper<Chain> for Mapper {
+    async fn to_block_stream_event(
+        &self,
+        logger: &Logger,
+        block_scoped_data: &BlockScopedData,
+    ) -> Result<Option<BlockStreamEvent<Chain>>, SubstreamsError> {
+        let step = ForkStep::from_i32(block_scoped_data.step).unwrap_or_else(|| {
+            panic!(
```

Comment:
I'm not sure we should be panicking here, mostly because this is something that could potentially happen in a production setting? Maybe just stopping the processing would be enough?